diff --git a/adapter/outbound/wireguard.go b/adapter/outbound/wireguard.go
index e31046b0..5a287652 100644
--- a/adapter/outbound/wireguard.go
+++ b/adapter/outbound/wireguard.go
@@ -128,6 +128,8 @@ type AmneziaWGOption struct {
 	RejectAfterTime        string `proxy:"reject-after-time,omitempty"`
 	KeepaliveTimeout       string `proxy:"keepalive-timeout,omitempty"`
 	MaxHandshakeAttempts   string `proxy:"max-handshake-attempts,omitempty"`
+	RandomTrailers         bool   `proxy:"random-trailers,omitempty"` // AmneziaWG v3.1+
+	DisableCookies         bool   `proxy:"disable-cookies,omitempty"` // AmneziaWG v3.1+
 }
 
 type wireguardDevice interface {
